AsyncDisplayKit/Details/ASPhotosFrameworkImageRequest.m

```diff
@@ -0,0 +1,169 @@
+//
+//  ASPhotosFrameworkImageRequest.m
+//  AsyncDisplayKit
+//
+//  Created by Adlai Holler on 9/25/15.
+//  Copyright © 2015 Facebook. All rights reserved.
+//
+
+#import "ASPhotosFrameworkImageRequest.h"
+#import "ASBaseDefines.h"
+
+NSString *const ASPhotosURLScheme = @"ph";
+
+static NSString *const _ASPhotosURLQueryKeyWidth = @"width";
+static NSString *const _ASPhotosURLQueryKeyHeight = @"height";
+
+// value is PHImageContentMode value
+static NSString *const _ASPhotosURLQueryKeyContentMode = @"contentmode";
+
+// value is PHImageRequestOptionsResizeMode value
+static NSString *const _ASPhotosURLQueryKeyResizeMode = @"resizemode";
+
+// value is PHImageRequestOptionsDeliveryMode value
+static NSString *const _ASPhotosURLQueryKeyDeliveryMode = @"deliverymode";
+
+// value is PHImageRequestOptionsVersion value
+static NSString *const _ASPhotosURLQueryKeyVersion = @"version";
+
+// value is 0 or 1
+static NSString *const _ASPhotosURLQueryKeyAllowNetworkAccess = @"network";
+
+static NSString *const _ASPhotosURLQueryKeyCropOriginX = @"crop_x";
+static NSString *const _ASPhotosURLQueryKeyCropOriginY = @"crop_y";
+static NSString *const _ASPhotosURLQueryKeyCropWidth = @"crop_w";
+static NSString *const _ASPhotosURLQueryKeyCropHeight = @"crop_h";
+
+@implementation ASPhotosFrameworkImageRequest
```

Comment:
Those are both very well-considered reasons to avoid Asset — thanks for the description!

Let's roll with the truly Objective-C "ASPhotosFrameworkURL" prefix.  I'm not confident that Components speaks to me as a developer that I should need to use this thing in order to load a Photos.framework image with ASMultiplexImageNode.

From the developer perspective, this is kinda like an "identifier", "specifier", or indeed "URL" — the actual object itself is serving as a mechanism to indicate a specific asset, and parameters associated with its loading.  We should try to eliminate people's consideration that on the backend, it generates an actual URL.  That API may even be best to leave private, but I won't press that point.

LoadingOptions, QueryParameters, this is such a hard name wtf!